### PR TITLE
Feature/US11143 authorization for fred report

### PR DESCRIPTION
**Pull request by silbermm**
Only show the authorized forms to the user

### CODE REVIEW

File: MinistryPlatform/Forms/IFormRepository.cs [hunk 1]
```diff
@@ -7,6 +7,6 @@ namespace MinistryPlatform.Forms
 {
     public interface IFormRepository
     {
-        IObservable<IEnumerable<MpForm>> GetFormNames();
-    }
+        IObservable<IEnumerable<MpForm>> GetFormNames();
```
Comment (vchtelmakh) [inline MinistryPlatform/Forms/IFormRepository.cs]: All that beautiful new whitespace.

File: MinistryPlatform/Models/MpForm.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@ namespace MinistryPlatform.Models
   public class MpForm
   {
 
+
```
Comment (vchtelmakh) [inline MinistryPlatform/Models/MpForm.cs]: Even more beautiful new whitespace!

File: crds-mp-tools.Tests/Controllers/FredResponsesReportControllerTest.cs [hunk 3]
```diff
@@ -20,12 +22,34 @@ namespace crds_mp_tools.Tests.Controllers
     public class FredResponsesReportControllerTest
     {
         private readonly Mock<IFredService> _fredService;
+        private readonly Mock<HttpContext> _httpContext;
         private readonly FredResponsesReportController _fixture;
-
+        private readonly Mock<ILogger<FredResponsesReportController>> _logger;
+        private const int userId = 2334;
```
Comment (vchtelmakh) [inline crds-mp-tools.Tests/Controllers/FredResponsesReportControllerTest.cs]: Whose userId is this?

**Comment by andycanterbury** [comment]
In CI/SQL/20171215_113500_US11143-Create-Roles_Forms-Subpage.sql it needs a USE Ministry Platform GO

File: MinistryPlatform.Tests/Forms/FormRepositoryTest.cs [hunk 4]
```diff
@@ -3,7 +3,6 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Moq;
 using MinistryPlatform.Forms;
-using System.Threading.Tasks;
```
Comment (andycanterbury) [inline MinistryPlatform.Tests/Forms/FormRepositoryTest.cs]: Yay for cleaning up usings!

File: MinistryPlatform/FormRoles/FormRolesRepository.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  {
+
+    private readonly IApiUserRepository _apiUserRepo;
+
+    public FormRolesRepository(IMinistryPlatformRestRepository ministryPlatformRestRepository, IConfigurationWrapper configurationWrapper, IApiUserRepository apiUserRepo) : base(ministryPlatformRestRepository, configurationWrapper)
+    {
+      _apiUserRepo = apiUserRepo;
+    }
+
+    public IObservable<IList<MpForm>> GetFormsInRole(string roleName)
```
Comment (andycanterbury) [inline MinistryPlatform/FormRoles/FormRolesRepository.cs]: I feel like this method and the plural version which takes a list of roles could be kind of combined. Either take the list and call this in a loop, or take the single role, put it in a list of 1 object and call that method. Could be more DRY.

File: crds-mp-tools.Tests/Controllers/FredResponsesReportControllerTest.cs (hunk 3 above)
Comment (silbermm) [reply crds-mp-tools.Tests/Controllers/FredResponsesReportControllerTest.cs]: not sure, but it's just a test

File: MinistryPlatform/FormRoles/FormRolesRepository.cs (hunk 5 above)
Comment (silbermm) [reply MinistryPlatform/FormRoles/FormRolesRepository.cs]: good suggestion, thanks! 👍

**Pull request merged**